drivers/wireless/lpwan/rn2483/rn2483.c

```diff
@@ -0,0 +1,862 @@
+  }
+  case RN2483_SET_SYNC:
+  {
+    written = snprintf(write_buffer, sizeof(write_buffer), "radio set sync %d\r\n", *(uint8_t *)data);
+    syslog(LOG_INFO, "%s", write_buffer);
+    // syslog(LOG_INFO, "Write Length: %d\n", written);
+    break;
+  }
+  default:
+  {
+    // Not a valid set command
+    syslog(LOG_INFO, "INVALID COMMAND GIVEN %d", cmd);
+    return -EINVAL;
+  }
+  }
+
+  ssize_t write_length = file_write(&priv->uart, write_buffer, written);
+  if (write_length < 0)
+  {
+    return write_length;
+  }
+
+  // Read
+  int buf_len = 50;
+  char read_buffer[buf_len];
+  size_t read = 0;
+  ssize_t fread;
+  char last = '\0';
+
+  // Read data into buffer until \r\n terminating sequence reached
+  while (read < buf_len)
+  {
+    fread = file_read(&priv->uart, &read_buffer[read], 1);
+    if (last == '\r' && read_buffer[read] == '\n')
+    {
+      break;
+    }
+    last = read_buffer[read];
+
+    if (fread < 0)
+      return fread;
+
+    read++;
+  }
+
+  // syslog(LOG_INFO, "%s", read_buffer);
+
+  if (strstr(read_buffer, "ok") != NULL)
+  {
+    syslog(LOG_INFO, "%s\n\n", SUCCESS);
+    return F_OK;
+  }
+  else if (strstr(read_buffer, "invalid_param") != NULL)
+  {
+    syslog(LOG_INFO, "%s\n\n", FAILURE);
+    return -EINVAL;
+  }
+  else
+  {
+    syslog(LOG_INFO, "INVALID VALUE (Not 'ok' or 'invalid_param')\n\n", FAILURE);
+    return -EINVAL;
+  }
+}
+
+int rn2483_set_config(FAR struct rn2483_dev_s *priv)
+{
+  int err;
+
+  /*  Set freq (Frequency in Hz)*/
+  err = rn2483_radio_set_cmd(priv, RN2483_SET_FREQ, &priv->config.freq);
+  if (err < 0)
+  {
+    return err;
+  }
+
+  /*  Set cr*/
+  err = rn2483_radio_set_cmd(priv, RN2483_SET_CR, &priv->config.cr);
+  if (err < 0)
+  {
+    return err;
+  }
+
+  /*  Set bw (Bandwidth in kHz)*/
+  err = rn2483_radio_set_cmd(priv, RN2483_SET_BW, &priv->config.bw);
+  if (err < 0)
+  {
+    return err;
+  }
+
+  /*  Set prlen (Preamble length)*/
+  err = rn2483_radio_set_cmd(priv, RN2483_SET_PRLEN, &priv->config.prlen);
+  if (err < 0)
+  {
+    return err;
+  }
+
+  /*  Set iqi (IQ invert, true for enabled)*/
+  err = rn2483_radio_set_cmd(priv, RN2483_SET_IQI, &priv->config.iqi);
+  if (err < 0)
+  {
+    return err;
+  }
+
+  /*  Set crc (Cyclic redundancy check, true for enabled)*/
+  err = rn2483_radio_set_cmd(priv, RN2483_SET_CRC, &priv->config.crc);
+  if (err < 0)
+  {
+    return err;
+  }
+
+  /*  Set pwr (Transmit output power)*/
+  err = rn2483_radio_set_cmd(priv, RN2483_SET_PWR, &priv->config.pwr);
+  if (err < 0)
+  {
+    return err;
+  }
+
+  /*  Set sf (Spread Factor)*/
+  err = rn2483_radio_set_cmd(priv, RN2483_SET_SF, &priv->config.sf);
+  if (err < 0)
+  {
+    return err;
+  }
+
+  /*  Set sync (Synchronization word)*/
+  err = rn2483_radio_set_cmd(priv, RN2483_SET_SYNC, &priv->config.sync);
+  if (err < 0)
+  {
+    return err;
+  }
+
+  /*  Set mod (Modulation: FSK or LoRa)*/
+  err = rn2483_radio_set_cmd(priv, RN2483_SET_MOD, &priv->config.mod);
```

Comment:
I'd set this early since I believe other radio parameters have different valid options based on which modulation is used. Double check this though.